embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/worker/ScheduledWorker.kt

```diff
@@ -43,4 +44,19 @@ internal class ScheduledWorker(
      * Submits a task for execution and returns a [Future].
      */
     fun submit(runnable: Runnable): Future<*> = impl.submit(runnable)
+
+    /**
+     * Submits a task for execution and returns a [Future].
+     */
+    fun <T> submit(callable: Callable<T>): Future<T> = impl.submit(callable)
+
+    @Deprecated(
+        "Use scheduleWithFixedDelay instead.",
+    )
+    fun scheduleAtFixedRate(
```

Comment:
Any reason we don't swap this out now?

Reply:
I don't want to touch the ANR detection logic at this time as it's fairly complex & the main purpose of this changeset was to consolidate thread creation in one place.